packages/node/src/protocol/utils/get-resolution-increments.ts

```diff
@@ -1,80 +1,74 @@
 ): Promise<{ [x: string]: BigNumber }> {
   const appInstance = stateChannel.getAppInstance(appInstanceId);
 
-  const appContract = new Contract(
+  const appDefinition = new Contract(
     appInstance.appInterface.addr,
     CounterfactualApp.abi,
     provider
   );
 
-  let attempts = 1;
-  let resolution: TransferTransaction = await appContract.functions.resolve(
-    appInstance.encodedLatestState,
-    appInstance.terms
+  const resolution = await appDefinition.functions.resolve(
+    appInstance.encodedLatestState
   );
 
-  // FIXME: This retry logic should apply to all view functions _and_ works for
-  //        arbitrary asset types. Presently it only works for ETH resolutions.
-  //        This was added to get the Playground demo launched sooner.
-  const wait = (ms: number) => new Promise(r => setTimeout(r, ms));
-  while (
-    bigNumberify(appInstance.terms.limit).gt(Zero) &&
-    resolution.value.every(v => v.eq(Zero)) &&
-    attempts < 10
-  ) {
-    log.info(`Empty resolution. Querying chain again. Attempt #${attempts}`);
+  /// todo(xuanji)
+  /// there used to be a bunch of retry logic here, to handle the case where
+  /// the appDefinition contract was deployed recently (and hence the provider
+  /// sometimes returning an incorrect definition). This TODO is to handle
+  /// that situation.
+
+  const resolveType = bigNumberify(await appDefinition.functions.resolveType());
 
-    resolution = await appContract.functions.resolve(
-      appInstance.encodedLatestState,
-      appInstance.terms
+  if (resolveType.eq(One)) {
+    /* ETHInterpreter */
+
+    const decoded = defaultAbiCoder.decode(
+      ["tuple(address,uint256)[]"],
+      resolution
     );
 
-    attempts += 1;
+    // todo(xuanji): check shape
+    const [[[address, to]]] = decoded;
 
-    await wait(1000 * attempts);
+    return { [address]: to };
   }
-  // END FIXME
+  if (resolveType.eq(Zero)) {
+    /* TWO_PARTY_OUTCOME */
 
-  if (resolution.assetType !== AssetType.ETH) {
-    return Promise.reject("Node only supports ETH resolutions at the moment.");
-  }
+    const [decoded] = defaultAbiCoder.decode(["uint256"], resolution);
 
-  return resolution.to.reduce(
-    (accumulator, currentValue, idx) => ({
-      ...accumulator,
-      [currentValue]: resolution.value[idx]
-    }),
-    {}
-  );
-}
+    const total = appInstance.limitOrTotal;
+    if (decoded.eq(Zero)) {
+      return {
+        [appInstance.beneficiaries[0]]: total
+      };
+    }
+    if (decoded.eq(One)) {
+      return {
+        [appInstance.beneficiaries[1]]: total
+      };
+    }
 
-export function getAliceBobMap(
-  channel: StateChannel
-): { alice: string; bob: string } {
-  return {
-    alice: channel.multisigOwners[0],
-    bob: channel.multisigOwners[1]
-  };
+    const i0 = total.div(2);
+    const i1 = total.sub(i0);
+    return {
+      [appInstance.beneficiaries[0]]: i0,
+      [appInstance.beneficiaries[1]]: i1
+    };
+  }
+  throw Error("unknown interpreter");
```

Comment:
The above logic would be nice with a switch statement.

Reply:
Not possible to do so directly since they are compared with `.eq` and not `==`